template/src/pdk.ts.ejs

```diff
@@ -12,6 +12,30 @@ function cast(caster: (v: any) => any, v: any): any {
   return caster(v)
 }
 
+export function mapFromJson<T>(obj: any, cast: (value: any) => T): Record<string, T> {
+  const result: Record<string, T> = {};
+
+  if (!obj) return result
+  
+  for (const [key, value] of Object.entries(obj)) {
+      result[key] = cast(value);
```

Comment:
I'm not sure it's safe to mutate this and we may need to copy. we'll be changing the type of a property that a user my still be holding right?

Reply:
We are mutating the local copy `result`